### DISCUSSION TITLE
Need help sorting dates

### DISCUSSION THREAD

**Discussion by gustavotrapp** [prisma/prisma#22891] [Q&A] [answered]
### Question

Hello guys, I need help sorting dates in prisma, currently where I work, we have our dates saved as strings (I don't know why either), I need help sorting them because currently I'm making a data table where you can sort and filter dates, so the lte and gte doesn't apply.

Preferably I'd like to use prismas `orderBy` but if that is impossible, any workaround would be good, thank you for your attention

NOTE: the row in trying to sort in the model below is `validade`

### Schema

```prisma
model documentos_legal_legislacao {
  id                Int      @id @default(autoincrement())
  titulo_documento String   @db.VarChar(255)
  numero_documento String   @db.VarChar(20)
  empresa_id        Int?
  empreendimento_id Int?
  responsavel_id    Int?
  validade          String  @db.VarChar(10)

  empresa           empresa? @relation(fields: [empresa_id], references: [id])
  empreendimento    empreendimento? @relation(fields: [empreendimento_id], references: [id])
  responsavel       auth_usuario? @relation(fields: [responsavel_id], references: [id])

  @@index([empresa_id], name: "empresa_id")
  @@index([empreendimento_id], name: "empreendimento_id")
  @@index([responsavel_id], name: "responsavel_id")
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @gustavotrapp 👋 

Thank you for raising this question.

Since your dates are stored as strings, the typical date comparison operators like lte (less than or equal to) and gte (greater than or equal to) won't work as expected just like you mentioned.

You might need to resort to using `queryRaw` to write a raw SQL query that sorts the dates stored as strings. Alternatively, parse `validade` into `DateTime` objects in your application code. Use JavaScript's Date.parse() or moment.js for robust parsing then sort based on these parsed objects within your application logic.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by gustavotrapp** [reply]
Thankyou, sadly, you answer confirmed my fear, of not being able to extract dates with any prisma API, which means that now I'll need to convert my prisma query into a rawquery to then concat it with the date query
